src/FNAPlatform/SDL2_FNAPlatform.cs

```diff
@@ -844,24 +844,27 @@ public static void RunLoop(Game game)
 					// Text Input
 					else if (evt.type == SDL.SDL_EventType.SDL_TEXTINPUT && !textInputSuppress)
 					{
-						string text;
-
 						// Based on the SDL2# LPUtf8StrMarshaler
 						unsafe
 						{
 							byte* endPtr = evt.text.text;
-							while (*endPtr != 0)
+							if (*endPtr != 0)
 							{
-								endPtr++;
+								int bytes = 0;
+								while (*endPtr != 0)
+								{
+									endPtr++;
+									bytes++;
+								}
+								// utf8 will never encode more characters than bytes in a string so bytes is a suitable upper estimate of buffersize needed
+								char* charsBuffer = stackalloc char[bytes];
+								int chars = Encoding.UTF8.GetChars(evt.text.text, bytes, charsBuffer, bytes);
+
+								for (int i = 0; i<chars; ++i)
```

Comment:
This probably has to get guarded too, and the NET40 compatible version can stick with the original `foreach`

Reply:
(Also spacing and use of `++` instead of `+= 1`